cmd/init.go

```diff
@@ -0,0 +1,556 @@
+package cmd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/amp-labs/cli/flags"
+	"github.com/amp-labs/cli/logger"
+	"github.com/amp-labs/cli/openapi"
+	"github.com/amp-labs/cli/request"
+	"github.com/manifoldco/promptui"
+	"github.com/spf13/cobra"
+	"sigs.k8s.io/yaml"
+)
+
+var ErrBadInput = errors.New("bad input")
+
+const (
+	NumProvidersShown = 10
+	YamlFileMode      = 0o644
+)
+
+func nonEmpty(fieldName string) func(string) error {
+	return func(s string) error {
+		if s == "" {
+			return fmt.Errorf("%w: %s cannot be empty", ErrBadInput, fieldName)
+		}
+
+		return nil
+	}
+}
+
+var initCmd = &cobra.Command{ //nolint:gochecknoglobals
+	Use:   "init",
+	Short: "create a new Ampersand project",
+	Long:  "Create a new Ampersand project.",
+	Run: func(cmd *cobra.Command, args []string) {
+		name, err := promptString("Name your integration", nonEmpty("Integration name"))
+		if err != nil {
+			logger.FatalErr("Unable to prompt for integration name", err)
+		}
+
+		provider, err := selectProvider(cmd.Context())
+		if err != nil {
+			logger.FatalErr("Unable to select provider", err)
+		}
+
+		disp := "my integration"
+
+		var integ openapi.Integration
+		integ.Name = name
+		integ.DisplayName = &disp
+		integ.Provider = provider.Name
+
+		if provider.Support.Proxy {
+			if err := setupProxy(&integ, provider); err != nil {
```

Comment:
For each of these `setUpRead` and `setUpWrite` functions, can you first ask the user if they'd like to set it up now or later? if they pick later, then skip the rest of the questions and add a block like this to the yaml file:

```
    read:
    # Define objects you'd like to read in this block
    # See https://docs.withampersand.com/define-integrations/read-actions
```

This will allow users the option to just create a skeletal file for now, and then fill it out later, otherwise it's a bit overwhelming to answer all these questions in one sitting the builder probably won't know all the answers yet without consulting docs, thinking through the integration with their team, etc.

Reply:
This one is gonna be pretty difficult to implement. I'm using `yaml.Marshal` to write the YAML, which doesn't have any way to hint about comments. I can do this but I'm basically gonna have to do my own custom marshaling, which might be fairly buggy. Are you sure you want to do this?

Reply:
Tell you what, I'll go ahead and merge this, and we can revisit this conversation in a separate issue. Since this is hidden it won't really impact anyone.